lib/XTR/StatsMigrationTask.php

```diff
@@ -0,0 +1,481 @@
+                    'type' => $stat['type'],
+                    'statDate' =>  $this->date->parse($stat['statDate'])->format('U'),
+                    'scheduleId' => (int) $stat['scheduleId'],
+                    'displayId' => (int) $stat['displayId'],
+                    'campaignId' => (int) $stat['campaignId'],
+                    'layoutId' => (int) $stat['layoutId'],
+                    'mediaId' => (int) $stat['mediaId'],
+                    'widgetId' => (int) $stat['widgetId'],
+                    'start' => $this->date->parse($stat['start'])->format('U'),
+                    'end' => $this->date->parse($stat['end'])->format('U'),
+                    'tag' => $stat['tag'],
+                    'duration' => isset($stat['duration']) ? (int) $stat['duration'] : $this->date->parse($stat['end'])->format('U') - $this->date->parse($stat['start'])->format('U'),
+                    'count' => isset($stat['count']) ? (int) $stat['count'] : 1,
+                ];
+
+                $watermark = $stat['statId'];
+
+                // Do the insert
+                $this->store->insert('INSERT INTO `stat` (' . $columns . ') VALUES (' . $values . ')', $params);
+                $this->store->commitIfNecessary();
+
+            }
+
+            // Give SQL time to recover
+            if ($watermark > 0) {
+                $this->log->debug('Stats migration effected '.$count.' rows, sleeping.');
+                sleep($options['pauseBetweenLoops']);
+            }
+        }
+
+    }
+
+    public function moveStatsFromStatArchiveToStatMongoDb($options = [])
+    {
+
+        $options = array_merge([
+
+            'numberOfRecords' => 10000,
+            'numberOfLoops' => 1000,
+            'pauseBetweenLoops' => 10,
+        ], $options);
+
+        // Migration from stat table to Mongo
+        $this->migrationStatToMongo($options);
+
+        // Migration from stat_archive table to Mongo
+        // After migration delete only stat_archive
+        if ($this->archiveExist == true) {
+            $this->migrationStatArchiveToMongo($options);
+        }
+    }
+
+    function migrationStatToMongo($options)
+    {
+        // Stat Archive Task
+        $archiveTask = $this->taskFactory->getByClass('\Xibo\XTR\\StatsArchiveTask');
+
+        $fileName = $this->config->getSetting('LIBRARY_LOCATION') . '.watermark_stat_mongo.txt';
+
+        // Get low watermark from file
+        $watermark = $this->getWatermarkFromFile($fileName, 'stat');
+
+        $sql = $this->store->getConnection()->prepare('SELECT statId FROM stat WHERE statId < :watermark ORDER BY statId DESC LIMIT 1');
+        $sql->bindParam(':watermark', $watermark, \PDO::PARAM_INT);
+        $sql->execute();
+
+        // Mark the Stats Archiver as disabled if there are records in stat table and set option archiveStats off
+        if ($sql->rowCount() > 0) {
+
+            // Quit the StatsArchiveTask if it is running
+            if ($archiveTask->runNow == 1) {
```

Comment:
runNow flag doesn't mean the task is running - it means that it will run next time the task scheduler runs, even if its not scheduled to run.

I think you want to check the `status` instead